extensions/amp-web-push/0.1/window-messenger.js

```diff
@@ -0,0 +1,383 @@
+/**
+ * Copyright 2017 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+ /*
+ * A Promise-based PostMessage helper to ease back-and-forth replies.
+ *
+ * This class is included separately a second time, by websites running
+ * amp-web-push, as well as by other push vendors, in a remote
+ * website's iframe. It should therefore keep external depenencies to a minimum,
+ * since this class must be transpiled to ES5 and "duplicated" outside of the
+ * AMP SDK bundle.
+ */
+import {getData} from '../../../src/event-helper';
+
+export class WindowMessenger {
+
+  /*
+   * Set debug to true to get console logs anytime a message is received,
+   * sent, or discarded.
+   */
+  constructor(options) {
```

Comment:
We can't mark constructors as private in Closure, I don't believe. Although I could be mistaken.